Feature for executing commands on server side

tortmayr opened the pull request:
This PR adds a sprotty feature extension which enables sprotty elements to act as command executor.
A commandExecutor has a command Id property. On double-click the commandExecutor sends a ExecuteServerCommandAction to the server.

* Implementation of the 'CommandExecutor" sprotty feature.
* Implementation and registration of the ExecuteServerCommand action on client/server side

planger commented on client/packages/glsp-sprotty/src/features/execute/execute-command.ts:
```diff
@@ -0,0 +1,32 @@
+
+export class ExecuteServerCommandAction implements Action {
+    static readonly KIND = "executeServerCommand"
+    kind = ExecuteServerCommandAction.KIND;
```
Wouldn't it be good if the command had an optional list of element ids to provide a context for the command execution?

tortmayr replied:
You are right. We should probably use a generic options map (<String,String>) though for passing additional arguments/context information etc.. Similar to how it'S handled in the RequestModelAction.

planger commented on client/packages/glsp-sprotty/src/features/execute/di.config.ts:
```diff
@@ -0,0 +1,19 @@
+import { ExecuteCommandMouseListener } from "./execute-command";
+
+const executeCommandModule = new ContainerModule(bind => {
+    bind(TYPES.MouseListener).to(ExecuteCommandMouseListener);
```
I'm wondering whether a double-click should invoke a command by default. I guess in most cases this will depend on the graphical language, could be a button, a link, Click with modifier, etc. A specific client could easily hook up its own listeners and invoke the ExecuteCommand. However, once we bind the double lick listener, clients won't be able to get rid of this listener easily. So this might be something that is rather bound by the specific client, isn't it?

tortmayr replied:
The modules are not added per default. Each specific client decides which feature-modules should be added to the sprotty di container. So the commandExecutonModule can easily be omitted or replaced.

see  also https://github.com/eclipsesource/graphical-lsp/blob/a2d16ed72f4c3ddf23b8064e8e5e5c87561a47d2/client/packages/workflow-sprotty/src/di.config.ts#L53

Of course we can extend the capabilites of the module later on to also react to button clicks etc. similiar to how it is handled in the expandModule  https://github.com/theia-ide/sprotty/blob/6ae2599560c24319a2ecc23c4ba8b0bdbb4258b4/client/src/features/expand/di.config.ts#L13

planger commented:
Oh, and one more comment, it would be good to integrate this functionality into the workflow example for demonstration purposes. Doesn't have to be part of this PR, but we should at least open an issue.

tortmayr commented:
@planger Thanks for the fast review. I addressed all your comments.
* The ExecuteServerCommandAction has an option map property for additional (context) parameters
* The CommandExectuion feature has been integrated into the workflow example. Double-clicking on a task icon will now trigger a simulation command on server side which outputs (i.e. logs) the simulated execution time of the task
